referenceframe/frame.go

```diff
@@ -31,14 +31,14 @@ type Limit struct {
 
 // RestrictedRandomFrameInputs will produce a list of valid, in-bounds inputs for the frame, restricting the range to
 // `lim` percent of the limits.
-func RestrictedRandomFrameInputs(m Frame, rSeed *rand.Rand, lim float64) []Input {
+func RestrictedRandomFrameInputs(m Frame, rSeed *rand.Rand, lim float64, nodePose []Input) []Input {
```

Comment:
`nodePose` is not a pose, but is just a set of Inputs, so should be named something differently.